test/Approximators/RangeApproximator/randsvd.jl

```diff
@@ -0,0 +1,334 @@
+            gram_matrix = approx_rec.U' * approx_rec.U
+            # check that the norm is 1, the diagonal is all 1
+            @test opnorm(gram_matrix) ≈ 1
+            # test that the diagonal is all ones
+            diag_gram_matrix = diag(gram_matrix)
+            for i = 1:compression_dim
+                @test diag_gram_matrix[i] ≈ 1
+            end
+
+            # bound from theorem 9.1 of Halko Martinsson and Tropp is satisfied
+            _,S,_ = svd(A)
+            norm(S[1:5] - approx_rec.S) < norm(S[6:end])^2 + 
+                norm(Diagonal(S[6:end]) * 
+                approx_rec.compressor.op[6:end, :] * 
+                pinv(approx_rec.compressor.op[1:5, :]))^2
+        end
+
+        # Test that compression_dim == n_rows gives a matrix that spans the range of A
+        # with power iterations, this way the approximation is exact
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 10,
+            cardinality = Right(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, false)
+            approx_rec = rapproximate(approx, A)
+            
+            @test typeof(approx_rec.compressor) == TestCompressorRecipe  
+            approx_rec.compressor.cardinality == Right()
+            @test approx_rec.power_its == 2
+            @test approx_rec.rand_subspace == false
+            @test approx_rec.n_rows == 10
+            @test approx_rec.n_cols == 10 
+            # Check that the matrix is orthogonal
+            gram_matrix = approx_rec.U' * approx_rec.U
+            # check that the norm is 1, the diagonal is all 1
+            @test opnorm(gram_matrix) ≈ 1
+            # test that the diagonal is all ones
+            diag_gram_matrix = diag(gram_matrix)
+            for i = 1:compression_dim
+                @test diag_gram_matrix[i] ≈ 1
+            end
+            
+            # Test that this spans the range and that the mul! function work
+            @test norm(A - approx_rec * (approx_rec' * A)) < ATOL
+            @test norm(A - A * approx_rec * approx_rec') < ATOL
+        end
+
+        # By testing the rapproximate function we also test rapproximate!
+        # with subspace iterations
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 5,
+            cardinality = Right(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, true)
+            approx_rec = rapproximate(approx, A)
+            
+            @test typeof(approx_rec.compressor) == TestCompressorRecipe  
+            # Check that the matrix is orthogonal
+            gram_matrix = approx_rec.U' * approx_rec.U
+            # check that the norm is 1, the diagonal is all 1s
+            @test opnorm(gram_matrix) ≈ 1
+            # test that the diagonal is all ones
```

Comment:
Move this to after the next test with the comment of testing the off diagonals near zero